frontend/app/src/MetricsManager.ts

```diff
@@ -90,7 +100,7 @@ export class MetricsManager {
   }
 
   public enqueue(evName: string, evData: Record<string, any> = {}): void {
-    if (!this.initialized || !this.sessionInfo.isSet) {
+    if (!this.initialized || !this.sessionInfo.isSet || !this.metricsUrl) {
```

Comment:
I think this could be a bit problematic in case the `this.metricsUrl` gets set to `undefined` / empty for the entire session. In that case, it might stack up a lot of events in `pendingEvents` that never get cleared up across the entire session. I think we can just remove the `|| !this.metricsUrl` here since the receiving of host config (and setting of the actual metrics URL) is guaranteed to happen before the session info (this.sessionInfo.isSet) is received.

Reply:
That makes sense, we check for a metrics URL before calling `track` anyway 👍🏼